libs/feature/search/src/lib/utils/mapper/elasticsearch.field.mapper.ts

```diff
@@ -31,9 +33,14 @@ type EsFieldMapperFn = (
 export class ElasticsearchFieldMapper {
   constructor(
     private metadataUrlService: MetadataUrlService,
-    private linkClassifier: LinkClassifierService
+    private linkClassifier: LinkClassifierService,
+    private translateService: TranslateService
   ) {}
 
+  private readonly lang3 = LANG_2_TO_3_MAPPER[this.translateService.currentLang]
```

Comment:
Is it a fixed variable ? With the language switcher, I expect people to dynamically change the language of the UI.

Reply:
I don't know what's the actual behavior  when we change locale.Is it reloading the page ? in this case it should work. 

If it doesn't reload, I'll get a bigger look on onLangChange to edit this variable.